### PR TITLE
dump controllerruntime's logs to klog

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
The default logs of controllerruntime is dummy, and errors will not be printed. It is unfriendly for developing.
https://github.com/karmada-io/karmada/blob/04365bd9f6219e211db0d647fe5614ee8d0fa6f3/vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go#L77-L85

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
update klog to `v2.60.1`, as same as [kubernetes v1.24.1](https://github.com/kubernetes/kubernetes/blob/v1.24.1/go.mod#L123). Which implements `logr.Logger`

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
> and errors will not be printed. It is unfriendly for developing.

which errors we are missing, can you give an example?

**Comment by ikaven1024** [comment]
> which errors we are missing, can you give an example?

errors returns in `Reconciler` is missing. e.g.
```go
package main

import (
	"context"
	"fmt"

	v1 "k8s.io/api/core/v1"
	controllerruntime "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/reconcile"
)

func main() {
	mgr, err := controllerruntime.NewManager(controllerruntime.GetConfigOrDie(), controllerruntime.Options{})
	_ = err
	reconciler := reconcile.Func(func(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
		return reconcile.Result{}, fmt.Errorf("!!!error in reconciler")
	})

	controllerruntime.NewControllerManagedBy(mgr).For(&v1.Namespace{}).Complete(reconciler)
	mgr.Start(context.TODO())
}
```
logs 
```
I0611 12:06:32.878604    6551 request.go:665] Waited for 1.047647583s due to client-side throttling, not priority and fairness, request: GET:https://10.211.55.4:5443/apis/policy/v1?timeout=32s
```

Then i add this in `import`
```go
_ "github.com/karmada-io/karmada/pkg/util/log"
```
logs:
```
I0611 12:10:59.931184    9297 request.go:665] Waited for 1.043018834s due to client-side throttling, not priority and fairness, request: GET:https://10.211.55.4:5443/apis/coordination.k8s.io/v1?timeout=32s
I0611 12:11:01.289008    9297 logr.go:249] "controller-runtime/metrics: Metrics server is starting to listen" addr=":8080"
I0611 12:11:01.289610    9297 internal.go:362] "Starting server" path="/metrics" kind="metrics" addr="[::]:8080"
I0611 12:11:01.289671    9297 controller.go:178] "controller/namespace: Starting EventSource" reconciler group="" reconciler kind="Namespace" source="kind source: *v1.Namespace"
I0611 12:11:01.289704    9297 controller.go:186] "controller/namespace: Starting Controller" reconciler group="" reconciler kind="Namespace"
I0611 12:11:01.395186    9297 controller.go:220] "controller/namespace: Starting workers" reconciler group="" reconciler kind="Namespace" worker count=1
E0611 12:11:01.395318    9297 controller.go:317] "controller/namespace: Reconciler error" err="!!!error in reconciler" reconciler group="" reconciler kind="Namespace" name="karmada-system" namespace=""
E0611 12:11:01.395382    9297 controller.go:317] "controller/namespace: Reconciler error" err="!!!error in reconciler" reconciler group="" reconciler kind="Namespace" name="kube-node-lease" namespace=""
E0611 12:11:01.395433    9297 controller.go:317] "controller/namespace: Reconciler error" err="!!!error in reconciler" reconciler group="" reconciler kind="Namespace" name="kube-public" namespace=""
E0611 12:11:01.395478    9297 controller.go:317] "controller/namespace: Reconciler error" err="!!!error in reconciler" reconciler group="" reconciler kind="Namespace" name="kube-system" namespace=""
```

File: cmd/webhook/main.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"k8s.io/component-base/logs"
 
 	"github.com/karmada-io/karmada/cmd/webhook/app"
+	_ "github.com/karmada-io/karmada/pkg/util/log"
```
Comment (XiShanYongYe-Chang) [inline cmd/webhook/main.go:10]: Is it also useful to import the package into a webhook?

Comment (ikaven1024) [reply cmd/webhook/main.go:10]: Roughly search in the code, find a log for webhook.
https://github.com/karmada-io/karmada/blob/0ec865ced6c8151c154c49c62883fcb12f839383/vendor/sigs.k8s.io/controller-runtime/pkg/manager/internal.go#L580-L582

Comment (XiShanYongYe-Chang) [reply cmd/webhook/main.go:10]: Thanks~

**Comment by RainbowMango** [comment]
I wonder if we can set the [Logger](https://github.com/karmada-io/karmada/blob/68069702930b14075299ed698c52331358018215/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go#L138) at somewhere like:
https://github.com/karmada-io/karmada/blob/68069702930b14075299ed698c52331358018215/cmd/controller-manager/app/controllermanager.go#L105

**Comment by ikaven1024** [comment]
> I wonder if we can set the [Logger](https://github.com/karmada-io/karmada/blob/68069702930b14075299ed698c52331358018215/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go#L138) at somewhere like:
> 
> https://github.com/karmada-io/karmada/blob/68069702930b14075299ed698c52331358018215/cmd/controller-manager/app/controllermanager.go#L105

Yes, options will override the global log. Both is ok for me. If you prefer latter, i will change to it.

**Comment by RainbowMango** [comment]
Thanks @ikaven1024 for the quick response.
I like the approach. Generally looks good to me.

Since we are still using k/k dependencies v1.23, let's wait for a while until we are going to update dependencies to v1.24(maybe next week, before that there are things to do.)

One question about this patch:
Does this only dumps `error` logs from the controller-runtime framework to klog? (log verbose concern)

**Comment by ikaven1024** [comment]
> Does this only dumps `error` logs from the controller-runtime framework to klog? (log verbose concern)

Besides error, it also have Info log with V()
image

**Comment by RainbowMango** [comment]
> Besides error, it also have Info log with V()

Can the logs dumped from the controller runtime be filtered by the log level defined with klog?
e..g 
cm.logger.V(4) would be ignored if we use `--v=2` in Karmada components.

**Comment by ikaven1024** [comment]
> Can the logs dumped from the controller runtime be filtered by the log level defined with klog

Yes. I run below with `--v=2`
```go
	controllerManager.GetLogger().V(2).Info("Log V2")
	controllerManager.GetLogger().V(3).Info("Log V3")
```
It prints
```
I0617 16:04:40.817312   20463 controllermanager.go:126] "Log V2"
```

**Comment by RainbowMango** [comment]
We are ready to go now. (In favor of #2050)

Thanks @ikaven1024

**Pull request merged**